src/App.jsx

```diff
@@ -1,16 +1,38 @@
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getTopArtists } from './Redux/artistActions';
 
 import Home from './components/Home';
 import Details from './components/Details';
+import Search from './components/Search';
 
 function App() {
+  const topArtists = useSelector((state) => state.artists.topArtists);
+  const artistName = useSelector((state) => state.artistName);
+  const artistListens = useSelector((state) => state.artsitListens);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (topArtists.length === 0) {
+      dispatch(getTopArtists());
+    }
+  }, [dispatch, topArtists.length]);
+
   return (
     <div className="App">
-      <BrowserRouter>
+      <BrowserRouter basename="/artist-metrics-webapp">
 
         <Routes>
-          <Route path="/home" element={<Home />} />
-          <Route path="/details" element={<Details />} />
+          <Route path="/" element={<Home />} />
+          <Route
+            path="details"
+            element={<Details artistName={artistName} artistListens={artistListens} />}
+          />
```

Comment:
Good job rendering the details page using this approach. However, there is a much simpler approach to doing this and this involves using one component as a placeholder for all details. The route of the component will look like this `<Route path="/item/:id" component={Item} />`. I strongly recommend using this approach because it will make your codebase much up to date with the new approach and with the project requirement.

HINT: add an event listener on each card and on click you can navigate to `./details:the id of the card clicked` and you can use `useParams` hook to read the data from the URL of the card clicked and you can then use that id to render the specific details.